Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -375,6 +463,152 @@ class SwiftObjectServerTests: SwiftSyncTestCase {
+                XCTAssertNotNil(john)
+                // Expect the server realm that overwrote the local realm to have had 1 object before client reset.
+                let results2 = after.objects(SwiftPerson.self)
+                XCTAssertEqual(results2.count, 1)
+                let paul2 = results2.filter("firstName == 'Paul'")
+                let john2 = results2.filter("firstName == 'John'")
+                XCTAssertFalse(paul2.isEmpty)
+                XCTAssert(john2.isEmpty)
+            }
+
+            var configuration = user.configuration(partitionValue: #function,
+                                                   clientResetMode: .discardLocal,
+                                                   notifyBeforeClientReset: beforeClientResetBlock,
+                                                   notifyAfterClientReset: afterClientResetBlock)
+            configuration.objectTypes = [SwiftPerson.self]
+
+            XCTAssertEqual(configuration.syncConfiguration?.clientResetMode, .discardLocal)
+            XCTAssertNotNil(configuration.syncConfiguration?.notifyBeforeClientReset)
+            XCTAssertNotNil(configuration.syncConfiguration?.notifyAfterClientReset)
+
+            // Seed 1 object, Upload to server
+            try autoreleasepool {
+                let realm = try Realm(configuration: configuration)
+                waitForDownloads(for: realm)
+                XCTAssertEqual(realm.objects(SwiftPerson.self).count, 0, "Precondition failure: Realm should be empty.") // Tear down may have not been run if test was cancelled.
+                try realm.write {
+                    realm.add(SwiftPerson(firstName: "Paul", lastName: "M"))
+                }
+                waitForUploads(for: realm)
+                XCTAssertEqual(realm.objects(SwiftPerson.self).count, 1)
+            }
+
+            // waitForUploads confirms the server received the upload request from the client
+            // It does not guarantee the uploaded objects are persisted to the backing datastore
+            // This expectation queries the server directly for the documents before continuing.
+            var documentCount = 0
+            var requestCount = 0
+            let timeBetweenRequests = 1
+            while documentCount < 1 {
+                collection.find(filter: [:]) { result in
+                    switch result {
+                    case .success(let documents):
+                        documentCount = documents.count
+                        if documentCount > 0 {
+                            XCTAssertEqual(documentCount, 1)
+                            XCTAssertEqual(documents[0]["firstName"]??.stringValue, "Paul")
+                            break
+                        }
+                    case .failure:
+                        XCTFail("Should find")
+                    }
+                }
+                if documentCount == 0 {
+                    if requestCount * timeBetweenRequests > 300 {
```

Comment:
Yes. This can take anywhere between 30 seconds and 4 minutes. Similar to what @ironage [experienced](https://github.com/realm/realm-core/blob/master/test/object-store/sync/sync_test_utils.cpp#L464).

I was previously using SwiftSyncHugeObject, which would take well beyond five minutes in some cases. So I switched to something smaller.

Reply:
I've seen the translator lag 10+ minutes behind. It's significantly slower than the sync server, and our sync tests are heavily bounded in speed by the sync server, so the translator ends up getting progressively further and further behind.

Disabling sync, deleting all docs via the mongo API, then re-enabling sync at the start of the test might be a net performance gain since it (might?) cancel all of the pending work from previous tests. If that doesn't work out we may need to launch a separate server process just for the client reset tests.